### DISCUSSION TITLE
Possible bug related to case preservation

### DISCUSSION THREAD

**Discussion by paj28** [jhy/jsoup#2144] [General]
I believe JSoup by default normalizes to lower case.

```
fun main() {
    val doc = Jsoup.parse("<foo BAR=1>")
    for (elem in doc.select("foo")) {
        for (attr in elem.attributes()) {
            println(attr.key)
        }
    }
}
```

Output: 
```
bar
```

But when the tag is svg, it does not:

```
fun main() {
    val doc = Jsoup.parse("<svg BAR=1>")
    for (elem in doc.select("svg")) {
        for (attr in elem.attributes()) {
            println(attr.key)
        }
    }
}
```
Output:
```
BAR
```
I'm not sure if this is a bug or some subtlety of intended behaviour that I've not quite understood.

**Comment by jhy** [comment]
Yes, it is a deliberate subtle behavior. SVG and MathML elements are specific "foreign" elements which do not get the HTML case normalization. They have specific casing. You can see the specific rules in https://html.spec.whatwg.org/multipage/parsing.html
